buildSrc/src/test/java/org/elasticsearch/gradle/NoticeTaskTest.java

```diff
@@ -0,0 +1,165 @@
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.gradle.testfixtures.ProjectBuilder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.rules.TemporaryFolder;
+
+import static java.util.Collections.singletonMap;
+import static org.apache.commons.io.FileUtils.readFileToString;
+import static org.apache.commons.io.FileUtils.write;
+import static org.junit.Assert.*;
+
+public class NoticeTaskTest extends GradleUnitTestCase {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+    private NoticeTask noticeTask;
+    private Project project;
+    final String outputHeader = "This is the header for the output file\nIt should contain:\n3 lines & 2 spaces";
+
+    public List<File> getListWithCopies() throws IOException{
+        File directory1 = temporaryFolder.newFolder("directoryA");
+        File directory2 = temporaryFolder.newFolder("directoryB");
+
+        File d1Notice = new File(directory1,"test-NOTICE.txt");
+        File d1NoticeCopy = new File(directory2,"test-NOTICE.txt");
+        File d1License = new File(directory1,"test-LICENSE.txt");
+        File d1LicenseCopy = new File(directory2,"test-LICENSE.txt");
+        File d2Notice = new File(directory2,"test2-NOTICE.txt");
+        File d2License = new File(directory2,"test2-LICENSE.txt");
+
+        write(d1Notice,"d1 Notice text file");
+        write(d1NoticeCopy,"d1 Copy Notice text file");
+        write(d2Notice,"d2 Notice text file");
+
+        write(d1License,"d1 License text file");
+        write(d1LicenseCopy,"d1 License text file");
+        write(d2License,"d2 License text file");
+
+
+        List<File> files = new ArrayList<>();
+        files.add(d1License);
+        files.add(d1Notice);
+        files.add(d1LicenseCopy);
+        files.add(d1NoticeCopy);
+        files.add(d2License);
+        files.add(d2Notice);
+
+        return files;
+    }
+
+    public List<File> getListWithoutCopies() throws IOException{
+        List<File> files = this.getListWithCopies();
+        files.remove(2);
+        files.remove(2);
+
+        return files;
+    }
+
+    @Test
+    public void verifyGenerateNoticeWithException()throws IOException{
+        // Setup everything so we can test out the task
+        project = createProject();
+        noticeTask = createTask(project);
+        File inputFile = new File(project.getProjectDir(),"NOTICE.txt");
+        File outputFile = new File(project.getProjectDir(),"OUTPUT.txt");
+
+        NoticeTask.writeToFile(inputFile,this.outputHeader,StandardCharsets.UTF_8);
+
+        // Set the input and output files on the NoticeTask so we can compare them later
+        noticeTask.setInputFile(inputFile);
+        noticeTask.setOutputFile(outputFile);
+        this.getListWithCopies().forEach(noticeTask::addLicenseDir);
+
+
+        try {
+            noticeTask.generateNotice();
```

Comment:
This won't fail if an exception is not thrown as it should. 
Use:
```
@Rule
    public ExpectedException expectedEx = ExpectedException.none();
```
instead.

Reply:
I updated to now expect a RuntimeException with a certain a certain error message.

Reply:
I think you should use the `ExpectedException` rule. 
I don't think this test passes as it's written now. 
The correct way would be to have a `fail()` as the last line in `try`, but the rule reads better.